src/java.base/share/classes/java/lang/invoke/StringConcatFactory.java

```diff
@@ -1580,7 +1580,8 @@ public void accept(CodeBuilder cb) {
                             cb.loadLocal(kind, cb.parameterSlot(i))
                               .invokestatic(CD_StringConcatHelper, "stringSize", methodTypeDesc);
                         }
-                        cb.iadd();
+                        cb.iadd()
+                          .invokestatic(CD_StringConcatHelper, "checkOverflow", MTD_int_int);
```

Comment:
Good, but wouldn't we generate slightly more compact code if this check was baked into the `stringSize` (when we call that) and only emitted in this generated for `char` parameters?